### ISSUE TITLE
reportUnknownVariableType when creating tuple of types (list, tuple)

### ISSUE THREAD

**Issue by evgenymarkov** [microsoft/pyright#5191]
**Describe the bug**

I'm trying to create tuple of types:

```python
list_types = (list, tuple)
```

Pyright (and Pylance) reports that type of "list_types" is partially unknown.

Also I can't assign (list, tuple) to less spicific type:

```python
list_types: Tuple[Type[Any], ...] = (list, tuple)
```

**Expected behavior**

I think that type should be:

```python
Tuple[Type[tuple[_T_co@tuple]], Type[list[_T@list]]]
```

Or something like this.

**Screenshots or Code**

<img width="766" alt="image" src="https://github.com/microsoft/pyright/assets/16443248/d1401228-182d-4245-8b39-6de455523c52">

**VS Code extension or command-line**

VS Code: 1.78.2
Pylance: v2023.5.40

**Comment by erictraut** [comment]
You appear to have enabled the `reportUnknownVariableType` diagnostic check. This check is off by default in "basic" type checking mode. Perhaps you've enabled "strict" mode?

This behavior is intended. 

You haven't provided a type annotation for the variable `list_types`, so its type is inferred. The inferred type of the expression `(list, tuple)` is `tuple[type[list[Unknown]], type[tuple[Unknown, ...]]]`.

You can address this using one of the following approaches:
1. Do not enable `reportUnknownVariableType` if you don't want this check to report errors.
2. Specify a type annotation: `list_types: tuple[type, ...] = (list, tuple)`.
3. Use a `# type: ignore` or `# pyright: ignore` comment to suppress the error.

> Also I can't assign (list, tuple) to less spicific type:

I'm not sure what you mean by this. Adding an annotation like this works for me.

> I think that type should be ...

No, it wouldn't make sense for type unsolved type variables to appear in the inferred type. These type variables would be out of scope, so there would be no way for them to be solved. A type argument of `Unknown` is correct here.
